src/store/reducers.js

```diff
@@ -19,13 +21,27 @@ export const todo = (state = [], { type, payload}) => {
           : todo
       )
     case 'DELETE_TODO':
-      const newState = Object.assign([], state)
-      const indexOfState = state.findIndex(todo => todo.id === payload.id)
+      return state.filter(todo => todo.id !== payload.id)
+    case 'SAVE_EDITING':
+      console.log('SAVE_EDITING')  
+        return state.map(todo =>
+        (todo.id === payload.id) 
+          ? {...todo, name: payload.name}
+          : todo
+        )
 
```

Comment:
Pode explicar melhor o que isso tá fazendo? você não deveria precisar alterar a maneira como o estado é passado pro componente só porque ele tem dois modos de exibição.
Se não for isso, desconsidere.

Reply:
Ele está editando o nome do todo, considerando que um objecto todo tem mais parametros que o nome da tarefa, ele está criando um todo e reescrevendo apenas o nome tipo

```javascript
let todoOne = {id: 1, name: 'antigo', createdAt:'2017-10-31'}

todoOne = {...todoOne, name: 'novo'}

// todoOne
{id: 1, name: 'novo', createdAt:'2017-10-31'}
```

Reply:
entendi